src/worker.js

```diff
@@ -286,6 +286,10 @@ self.onmessage = function(e) {
       if (Module['_pthread_self']()) { // If this thread is actually running?
         Module['_emscripten_current_thread_process_queued_calls']();
       }
+    } else if (e.data.cmd === 'processProxyingQueue') {
+      if (Module['_pthread_self']()) { // If this thread is actually running?
+        Module['_emscripten_proxy_execute_queue'](e.data.queue);
+      }
```

Comment:
This part won't DCE away if user does not utilize this proxying queue.

This might be worth adding a `#if LIBRARY_PROXY_QUEUE` preprocessor that gets enabled when the user is targeting the new proxy queue mechanism.

Reply:
IIUC, there wouldn't be any benefit to having that macro once we start using this for the system queue as well, right?

Reply:
I think you can wrap this in `hasExportedFunction('__emscripten_proxy_execute_queue')` which will have the effect of only including it when needed. (note the double underscores)